File: submissions/1666-make-the-string-great/solution.ts
function makeGood(s: string): string {
  let result = [];

    for (let i = 0; i < s.length; i++) {
        const current = s[i];
        const last = result.length ? result[result.length - 1] : '';

        if (current.toLowerCase() === last.toLowerCase() && current !== last) {
            result.pop();
            continue;
        }

        result.push(current);
    }

    return result.join('');
};

File: submissions/1666-make-the-string-great/solution.py
class Solution:
    def makeGood(self, s: str) -> str:
        stack = []
        
        for c in s:
            if stack: 
                if c.lower() == stack[-1] and c != stack[-1]  or c.upper() == stack[-1]  and  c != stack[-1]:
                    stack.pop()
                else: 
                    stack.append(c)
            else:
                stack.append(c)
        
        return "".join(stack)
